pkg/envoy/xds/ack_test.go

```diff
@@ -229,3 +229,98 @@ func (s *AckSuite) TestDeleteMultipleNodes(c *C) {
 	// The resource name is ignored. For delete, we only consider the version.
 	c.Assert(comp, IsCompleted)
 }
+
+func (s *AckSuite) TestRevertInsert(c *C) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	typeURL := "type.googleapis.com/envoy.api.v2.DummyConfiguration"
+	wg := completion.NewWaitGroup(ctx)
+
+	cache := NewCache()
+	acker := NewAckingResourceMutatorWrapper(cache, IstioNodeToIP)
+
+	// Create version 1 with resource 0.
+	comp := wg.AddCompletion()
+	defer comp.Complete()
+
+	// Insert.
+	revert := acker.Upsert(typeURL, resources[0].Name, resources[0], []string{node0}, comp)
```

Comment:
Can you add a test that runs `upsert` 2 times and you want to revert the first upsert?